TCAT/Controllers/AllStopsTableViewController.swift

```diff
@@ -64,42 +64,24 @@ class AllStopsTableViewController: UIViewController {
     }
 
     private func createSectionIndexesForBusStop() {
-        var currentChar: Character?
         var sectionIndexDictionary: [String: [Place]] = [:]
-        var currBusStopArray: [Place] = []
+        var numberBusStops: [Place] = []
 
-        currentChar = allStops.first?.name.capitalized.first
-
-        var numberBusStops: [Place] = {
-            if let firstStop = allStops.first {
-                return [firstStop]
-            }
-            return []
-        }()
-
-        if currentChar != nil {
+        if !allStops.isEmpty {
             for busStop in allStops {
-                if let firstChar = busStop.name.capitalized.first {
-                    if currentChar != firstChar {
-                        if !CharacterSet.decimalDigits.contains(currentChar!.unicodeScalars.first!) {
-                            sectionIndexDictionary["\(currentChar!)"] = currBusStopArray
-                            currBusStopArray = []
-                        }
-                        currentChar = firstChar
-                        currBusStopArray.append(busStop)
+                if let firstChar = busStop.name.capitalized.first, let firstScalar = firstChar.unicodeScalars.first {
+                    if CharacterSet.decimalDigits.contains(firstScalar) {
+                        numberBusStops.append(busStop)
                     } else {
-                        if CharacterSet.decimalDigits.contains(currentChar!.unicodeScalars.first!) {
-                            numberBusStops.append(busStop)
+                        if sectionIndexDictionary["\(firstChar)"] == nil {
+                            sectionIndexDictionary["\(firstChar)"] = [busStop]
                         } else {
-                            currBusStopArray.append(busStop)
+                            sectionIndexDictionary["\(firstChar)"]?.append(busStop)
```

Comment:
nit to take out optional chaining: 
```swift
if let stops = sectionIndexDictionary["\(firstChar)"] {
  stops.append(busStop)
} else {
  sectionIndexDictionary["\(firstChar)"] = [busStop]
}
```

Reply:
if let makes stops immutable so should I do if var

Reply:
Yea if var, my b